python/apigw-lambda-rds/app.py

```diff
@@ -0,0 +1,76 @@
+import os
+from aws_cdk import (
+    aws_apigateway as apigateway,
+    aws_cloudformation as cfn,
+    aws_events as events, 
+    aws_lambda as lambda_, 
+    aws_events_targets as targets,
+    aws_ec2 as ec2,
+    aws_rds as rds,
+    aws_iam as iam,
+    core,
+)
+
+
+class ApiGwLambdaRds(core.Stack):
+    def __init__(self, scope: core.Construct, id: str, default_db_name: str, **kwargs) -> None:
+        
+        super().__init__(scope, id, *kwargs)
+
+        vpc = ec2.Vpc(
+            self, 'MyVpc',
+            cidr='192.168.123.0/24',
+            max_azs=2
+        )
+
+        rds_instance = rds.DatabaseInstance(
+            self, 'MyRDSInstance',
+            engine=rds.DatabaseInstanceEngine('Mysql', rds.SecretRotationApplication('MysqlRotationSingleUser','1')),
+            master_username='admin',
+            instance_class=ec2.InstanceType('t2.small'),
+            database_name='Transport',
+            vpc=vpc
+        )
+        rds_instance._security_group.add_ingress_rule(ec2.Peer.ipv4(cidr_ip='192.168.123.0/24'), ec2.Port.tcp(3306))
```

Comment:
is this usage the reason the defaults aren't used for the cidr_block? 

or do the defaults not work for some reason?